src/zod.ts

```diff
@@ -280,3 +284,122 @@ export const Abi = z.array(
     ),
   ]),
 )
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Typed Data Types
+
+export const TypedDataDomain = z.object({
+  chainId: z.number().optional(),
+  name: z.string().regex(/[a-zA-Z$_][a-zA-Z0-9$_]*/).optional(),
```

Comment:
Might be worth pulling out `z.string().regex(/[a-zA-Z$_][a-zA-Z0-9$_]*/)` and updating a few other places, like `AbiParameter.name`, `AbiFunction.name`, etc.

Reply:
I agree. Will update accordingly